addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-            domain = [('date', '<=', st.date), ('journal_id', '=', st.journal_id.id)]
-            # The reason why we have to perform this test is because we have two use case here:
-            # First one is in case we are creating a new record, in that case that new record does
-            # not have any id yet. However if we are updating an existing record, the domain date <= st.date
-            # will find the record itself, so we have to add a condition in the search to ignore self.id
-            if not isinstance(st.id, models.NewId):
-                domain.extend(['|', '&', ('id', '<', st.id), ('date', '=', st.date), '&', ('id', '!=', st.id), ('date', '!=', st.date)])
-            previous_statement = self.search(domain, limit=1, order='date desc, id desc')
-            st.previous_statement_id = previous_statement.id
-
-    name = fields.Char(string='Reference', states={'open': [('readonly', False)]}, copy=False, readonly=True)
-    reference = fields.Char(string='External Reference', states={'open': [('readonly', False)]}, copy=False, readonly=True, help="Used to hold the reference of the external mean that created this statement (name of imported file, reference of online synchronization...)")
-    date = fields.Date(required=True, states={'confirm': [('readonly', True)]}, index=True, copy=False, default=fields.Date.context_today)
-    date_done = fields.Datetime(string="Closed On")
-    balance_start = fields.Monetary(string='Starting Balance', states={'confirm': [('readonly', True)]}, compute='_compute_starting_balance', readonly=False, store=True, tracking=True)
-    balance_end_real = fields.Monetary('Ending Balance', states={'confirm': [('readonly', True)]}, compute='_compute_ending_balance', recursive=True, readonly=False, store=True, tracking=True)
-    state = fields.Selection(string='Status', required=True, readonly=True, copy=False, tracking=True, selection=[
-            ('open', 'New'),
-            ('posted', 'Processing'),
-            ('confirm', 'Validated'),
-        ], default='open',
-        help="The current state of your bank statement:"
-             "- New: Fully editable with draft Journal Entries."
-             "- Processing: No longer editable with posted Journal entries, ready for the reconciliation."
-             "- Validated: All lines are reconciled. There is nothing left to process.")
+    @api.depends('line_ids', 'line_ids.date')
+    def _compute_date(self):
+        for statement in self:
+            statement.date = bool(statement.line_ids) and max(statement.line_ids.mapped('date'))
+
+    @api.depends('line_ids.journal_id')
+    def _compute_journal_id(self):
+        for statement in self:
+            statement.journal_id = statement.line_ids.journal_id[:1]
+
+    @api.depends('is_valid_balance_end',)
+    def _compute_state(self):
+        for statement in self:
+            if statement.state == 'posted':
+                # we do not change the state of a posted statement automatically
```

Comment:
is it supposed to be the `validated` state? So editing a statement line of a validated statement is ok... It looks weird to me

Reply:
It is still new, I am fixing it all over the code and view